pennylane/devices/qubit/sampling.py

```diff
@@ -353,3 +394,64 @@ def sample_state(
     powers_of_two = 1 << np.arange(num_wires, dtype=np.int64)[::-1]
     states_sampled_base_ten = samples[..., None] & powers_of_two
     return (states_sampled_base_ten > 0).astype(np.int64)
+
+
+# pylint:disable = unused-argument
+def _sample_state_jax(
+    state,
+    shots: int,
+    is_state_batched: bool = False,
+    wires=None,
+    prng_key=None,
+) -> np.ndarray:
+    """
+    Returns a series of samples of a state for the JAX interface based on the PRNG if provided.
+    If one is not provided, a random PRNG key will be generated.
+
+    Args:
+        state (array[complex]): A state vector to be sampled
+        shots (int): The number of samples to take
+        is_state_batched (bool): whether the state is batched or not
+        wires (Sequence[int]): The wires to sample
+        prng_key (Union[int, jax.random.PRNGKey]): A``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. You should only end up here
+            if there is a PRNG key.
+
+    Returns:
+        ndarray[int]: Sample values of the shape (shots, num_wires)
+    """
+    # pylint: disable=import-outside-toplevel
+    import jax
+    import jax.numpy as jnp
+
+    key = prng_key
+    if isinstance(prng_key, int):
+        key = jax.random.PRNGKey(prng_key)
+
+    total_indices = len(state.shape) - is_state_batched
+    state_wires = qml.wires.Wires(range(total_indices))
+
+    wires_to_sample = wires or state_wires
+    num_wires = len(wires_to_sample)
+    basis_states = np.arange(2**num_wires)
+
+    probs = qml.probs(wires=wires_to_sample).process_state(state, state_wires)
+
+    if is_state_batched:
+        # Produce separate keys for each of the probabilities along the broadcasted axis
+        keys = []
+        for _ in state:
+            key, subkey = jax.random.split(key)
+            keys.append(subkey)
+        samples = jnp.array(
+            [
+                jax.random.choice(_key, basis_states, shape=(shots,), p=prob)
+                for _key, prob in zip(keys, probs)
+            ]
+        )
+    else:
+        samples = jax.random.choice(key, basis_states, shape=(shots,), p=probs)
```

Comment:
this is the only section that's different (plus the key handling) - could you change the function to only be this (something like `generate_samples_jax`) and have the non-jax counterpart in a different function (like `generate_samples`) so we don't have to copy-paste the before/after stuff. I'm only asking because sometimes the before/after stuff changes, and I don't want it to be missed in the jax one (eg. I changed the type from bool8 to int64 a while back)

ok after looking at tests I realize that they might need some re-writing, so we can just do this later, but still wanna mention it.

Reply:
Makes sense, I will add a shortcut story